packages/core/useScriptTag/index.ts

```diff
@@ -0,0 +1,142 @@
+   * @returns Promise<HTMLScriptElement>
+   */
+  const load = (waitForScriptLoad = true): Promise<HTMLScriptElement|boolean> =>
+    new Promise((resolve, reject) => {
+      // Some little closure for resolving the Promise.
+      const resolveWithElement = (el: HTMLScriptElement) => {
+        scriptTag.value = el
+        resolve(el)
+        return el
+      }
+
+      // Check if document actually exists, otherwise resolve the Promise (SSR Support).
+      if (!document) {
+        resolve(false)
+        return
+      }
+
+      // Local variable defining if the <script> tag should be appended or not.
+      let shouldAppend = false
+
+      let el: HTMLScriptElement = <HTMLScriptElement>(
+        document.querySelector(`script[src="${src}"]`)
+      )
+
+      // Script tag found, preparing the element for appending
+      if (!el) {
+        // Create element and set required attributes
+        el = document.createElement('script')
+        el.type = type
+        el.async = async
+        el.src = isString(src) ? src : src.value
+        // Optional attributes
+        if (defer) el.defer = defer
+        if (crossOrigin) el.crossOrigin = crossOrigin
+        if (noModule) el.noModule = noModule
+        if (referrerPolicy) el.referrerPolicy = referrerPolicy
+        // Enables shouldAppend
+        shouldAppend = true
+      }
+      // Script tag already exists, resolve the loading Promise with it.
+      else if (el.hasAttribute('data-loaded')) {
+        resolveWithElement(el)
+      }
+
+      // Event listeners
+      el.addEventListener('error', event => reject(event))
+      el.addEventListener('abort', event => reject(event))
+      el.addEventListener('load', () => {
+        el.setAttribute('data-loaded', 'true')
+
+        if (onLoaded) onLoaded(el)
+
+        resolveWithElement(el)
+      })
+
+      // Append the <script> tag to head.
+      if (shouldAppend) el = document.head.appendChild(el)
+
+      // If script load awaiting isn't needed, we can resolve the Promise.
+      if (!waitForScriptLoad) resolveWithElement(el)
+    })
+
+  /**
+   * Unload the script specified by `src`.
+   *
+   * @returns Promise<boolean>
+   */
+  const unload = (): Promise<boolean> =>
+    new Promise((resolve) => {
+      // Check if document actually exists, otherwise reject the Promise.
+      if (!document) {
+        resolve(false)
+        return
+      }
+
+      // Check if script tag actually exists, otherwise resolve the Promise.
+      if (!scriptTag.value) {
+        resolve(true)
+        return
+      }
+
+      // Remove the <script> tag from the document head.
+      document.head.removeChild(scriptTag.value)
+
+      // Reset the <script> tag reference.
+      scriptTag.value = undefined
+
+      resolve(true)
+    })
+
+  // Try to load the script onMounted.
+  tryOnMounted(async() => {
+    if (immediate) await load()
+  })
```

Comment:
```suggestion
  if (immediate)
    tryOnMounted(load())
```

Reply:
```
  // Try to load the script onMounted.
  if (immediate) tryOnMounted(async() => load())

  // Try to unload the script onUnMounted.
  tryOnUnmounted(async() => unload())
```

I came up with this, using "load()" as tryOnMounted parameter does not match the function signature.

I think we could update the "Fn" type to also allow promises, or create a new type instead.

This should be the subject of another PR / issue though.

Thanks again for your review.